python/erdos/operator.py

```diff
@@ -63,7 +65,7 @@ def save_trace_events(self, file_name):
             json.dump(self._trace_events, write_file)
 
 
-class Source(BaseOperator):
+class Source(BaseOperator[T]):
```

Comment:
```suggestion
class Source(BaseOperator, Generic[T]):
```

Use multiple inheritance to make each operator class generic.